ansible/roles/proxy/tasks/main.yml

```diff
@@ -1,5 +1,8 @@
 ---
 
+  - name: Copy sources list.
+    copy: src=sources.list dest=/etc/apt/sources.list owner=root group=root mode=0640
+
   - name: Copy hosts file.
     template: src=hosts.j2 dest=/etc/hosts backup=no owner=root mode=0750
 
```

Comment:
I think you should add `update_cache=yes` in the apt squid proxy task, to parse the new sources.list.